src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+        )
+        for i in range(0, len(quadratic_filter_array)):
+            quadratic_filter_array[i] = quadratic_filter_array[i][1:-6]
+
+        # Final Downsample by 5
+        for i, ind in enumerate([1, 4, 3, 3, 3]):
+            quadratic_filter_array[i] = quadratic_filter_array[i][ind::5]
+
+    return quadratic_filter_array, messages
+
+
+def wavelet_transform(
+    signal: np.ndarray,
+    quadratic_filter_array: np.ndarray,
+) -> np.ndarray:
+    """
+    Calculates the wavelet transform of a signal using quadratic spline wavelet.
+    It calculates wavelets in scales of 1 to 4.
+
+    Args:
+        signal (np.ndarray): The target signal designated for filtering.
+        quadratic_filter_array (np.ndarray): Qquadratic spline filter bank.
+
+    Returns:
+        wavelet_transform_matrix (np.ndarray): Resultant Wavelet Transform Matrix.
+    """
+    if signal.size == 0:
+        raise ValueError("Input is an empty array.")
+    if not isinstance(quadratic_filter_array, list) or any(
+        arr.size == 0 for arr in quadratic_filter_array
+    ):
+        raise ValueError(
+            "quadratic_filter_array should be a list of 1D numpy arrays."
+        )
+    if signal.squeeze().ndim > 1 or any(
+        not isinstance(arr, np.ndarray) or arr.ndim != 1
+        for arr in quadratic_filter_array
+    ):
+        raise ValueError("Input is a higher-dimensional array, not a matrix.")
+
+    wavelet_transform_matrix = np.zeros((len(signal), 5))
+    for i, filt in enumerate(quadratic_filter_array):
+        wavelet_transform_matrix[:, i] = np.transpose(lfilter(filt, 1, signal))
+    return wavelet_transform_matrix
+
+
+def create_filter(messages: dict) -> dict:
+    """
+    Creates quadratic spline filters using a variation on
+    Mallat's Algorithm, algorithme à trous.
+
+    Args:
+        messages (dict): Dictionary containing information relevant to the algorithm.
+
+    Returns:
+        quadratic_filter_dictionary (dict): A dictionary relevant to the algorithm filter-bank.
+    """
+    quadratic_filter_array, messages = quadratic_splines_filterbank(
+        messages["setup"]["wavedet"]["freq"], messages
+    )
+    lengths = [len(q) for q in quadratic_filter_array]
+    decimation_values = [int(np.floor((l - 1) / 2)) for l in lengths]
+    quadratic_filter_dictionary = {
+        "filters": quadratic_filter_array,
+        "lengths": lengths,
+        "d_values": decimation_values,
+    }
+    return quadratic_filter_dictionary
+
+
+def filter_segment_of_signal(
+    quadratic_filter_dictionary: dict,
+    current_samp: int,
+    initial_samp: int,
+    num_samps: int,
+    segment_boundaries: list,
+    signal_segment: np.ndarray,
+    messages: dict,
+):
+    """
+    Uses a Wavelet Transform on individual ECG signal excerpts, after
+    Generating quadratic-spline filter banks.
+
+    Args:
+        quadratic_filter_dictionary (dict): A dictionary relevant to the algorithm filter-bank.
```

Comment:
The `quadratic_filter_dictionary` is rather cryptic, I'd prefer simply to have each of it's values as an independent variable